scripts/rotate-aws-creds/common.sh

```diff
@@ -0,0 +1,132 @@
+    rm "${TEMP_DIR}/*.CredentialsRequest.json" 2>/dev/null || true
+
+    # get original credentialsrequests and delete them
+    for NS in $(KUBECONFIG=$KUBECONFIG $OC get credentialsrequests --all-namespaces -o json | jq -r '.items[].metadata.namespace' | uniq | grep -e ^openshift -e ^kube -e ^default$ -e ^redhat | grep -v openshift-cloud-credential-operator);
+    do
+        for CR in $(KUBECONFIG=$KUBECONFIG $OC get credentialsrequests -n "$NS" -o json | jq -r '.items[].metadata.name');
+        do
+            # filename to stash resource in
+            F=${TEMP_DIR}/${CR}_${NS}.CredentialsRequest.json
+
+            # get the original json and clean it up
+            KUBECONFIG=$KUBECONFIG $OC -n "$NS" get credentialsrequests "$CR" -o json | jq -r 'del(.status) | del(.metadata.annotations) | del(.metadata.creationTimestamp) | del(.metadata.finalizers) | del(.metadata.generation) | del(.metadata.ownerReferences) | del(.metadata.resourceVersion) | del(.metadata.selfLink) | del(.metadata.uid)' > "$F"
+
+            # delete the original resource
+            KUBECONFIG=$KUBECONFIG $OC -n "$NS" delete credentialsrequests "$CR"
+
+            # recreate resource
+            KUBECONFIG=$KUBECONFIG $OC create -f "$F"
+            # name of referenced secret
+            S=$(jq -r '.spec.secretRef.name' "$F")
+
+            # cleanup any pods that referenced the secret (note we're ignoring namespace, exceptionally small risk)
+            IFS_OLD=$IFS
+            IFS=
+            for X in $(KUBECONFIG=$KUBECONFIG $OC get pods --all-namespaces -o json | jq -r ".items[] | select(.spec.volumes[].secret.secretName == \"$S\") | \"$OC -n \" + .metadata.namespace + \" delete pod \" + .metadata.name");
+            do
+                X="KUBECONFIG=$KUBECONFIG $X"
+                # do not wait, just move on
+                eval "$X" &
+            done
+            IFS=$IFS_OLD
+
+            info "Rotated SRE managed CredentialsRequest '$CR' in '$NS'."
+        done
+    done
+}
+
+iam_user_create_accesskey() {
+    local AWS_USERNAME=$1
+
+    info "IAM User: $AWS_USERNAME"
+
+    local ORIG_ACCOUNTID
+    ORIG_ACCOUNTID=$(AWS_PROFILE=$AWS_PROFILE aws sts get-caller-identity | jq -r '.Account')
+
+    info "Account ID: $ORIG_ACCOUNTID"
+
+    # create a new access key
+    local ACCESS_KEY_JSON
+    ACCESS_KEY_JSON=$(AWS_PROFILE=$AWS_PROFILE aws iam create-access-key --user-name "$AWS_USERNAME")
+
+    # get encoded access key data
+    NEW_ACCESS_KEY_ID=$(echo "$ACCESS_KEY_JSON" | jq -r '.AccessKey.AccessKeyId' | tr -d '\n')
+    ENCODED_ACCESS_KEY_ID=$(echo -n "$NEW_ACCESS_KEY_ID" | base64)
+    NEW_SECRET_ACCESS_KEY=$(echo "$ACCESS_KEY_JSON" | jq -r '.AccessKey.SecretAccessKey' | tr -d '\n')
+    ENCODED_SECRET_ACCESS_KEY=$(echo -n "$NEW_SECRET_ACCESS_KEY" | base64)
+
+    info "Created new AccessKey for IAM User '$AWS_USERNAME'."
+
+    # verify it works (need to sleep a bit)
+    echo "Waiting 10 seconds for new AccessKey to activate..."
```

Comment:
I hate AWS' eventual consistency. This was a _big_ issue with the aws-account-operator